src/Bicep.Core/Semantics/Namespaces/SystemNamespaceType.cs

```diff
@@ -541,67 +552,125 @@ public static class SystemNamespaceType
                 return LanguageConstants.String;
             };
 
-        private static TypeSymbol LoadTextContentTypeBuilder(IBinder binder, IFileResolver fileResolver, IDiagnosticWriter diagnostics, ImmutableArray<FunctionArgumentSyntax> arguments, ImmutableArray<TypeSymbol> argumentTypes)
+        private static FunctionResult LoadTextContentResultBuilder(IBinder binder, IFileResolver fileResolver, IDiagnosticWriter diagnostics, ImmutableArray<FunctionArgumentSyntax> arguments, ImmutableArray<TypeSymbol> argumentTypes)
         {
-            if (argumentTypes[0] is not StringLiteralType filePathType)
+            return TryLoadTextContentFromFile(binder, fileResolver, diagnostics,
+                (arguments[0], argumentTypes[0]),
+                arguments.Length > 1 ? (arguments[1], argumentTypes[1]) : null,
+                out var fileContent,
+                LanguageConstants.MaxLiteralCharacterLimit)
+                ? new(new StringLiteralType(fileContent))
+                : new(LanguageConstants.String);
+        }
+
+        private static FunctionResult LoadJsonContentResultBuilder(IBinder binder, IFileResolver fileResolver, IDiagnosticWriter diagnostics, ImmutableArray<FunctionArgumentSyntax> arguments, ImmutableArray<TypeSymbol> argumentTypes)
+        {
+            string? tokenSelectorPath = null;
+            if (arguments.Length > 1)
+            {
+                if (argumentTypes[1] is not StringLiteralType tokenSelectorType)
+                {
+                    diagnostics.Write(DiagnosticBuilder.ForPosition(arguments[1]).CompileTimeConstantRequired());
+                    return new(LanguageConstants.Any);
+                }
+                tokenSelectorPath = tokenSelectorType.RawStringValue;
+            }
+            if (!TryLoadTextContentFromFile(binder, fileResolver, diagnostics,
+                    (arguments[0], argumentTypes[0]),
+                    arguments.Length > 2 ? (arguments[2], argumentTypes[2]) : null,
+                    out var fileContent,
+                    LanguageConstants.MaxJsonFileCharacterLimit))
+            {
+                return new(LanguageConstants.Any);
+            }
+
+            if (fileContent.TryFromJson<JToken>() is not { } token)
```

Comment:
We should avoid using `Newtonsoft.Json` for new features. Can you replace it with `System.Text.Json`? The Bicep.Core project contains some `System.Text.Json` helpers that maybe helpful, for example:
- [Factory to create `JsonElement` instances](https://github.com/Azure/bicep/blob/main/src/Bicep.Core/Json/JsonElementFactory.cs#L38)
- [Extension method to select a json path](https://github.com/Azure/bicep/blob/main/src/Bicep.Core/Extensions/JsonElementExtensions.cs#L99)

Reply:
System.text.json does not support JsonPath queries (https://github.com/dotnet/runtime/issues/31068) and I'd like to avoid implementing my own system.

Reply:
I started working on reimplementing this to use your extension methods but the main difference is that newtonsoft.json returns me single token for a match, while here I get list of JsonElements. Additionally, it's quite hard IMHO to work with System.Text.Json - i.e. no easy distinction if we have an integer or floating point number. 
Also, there were already some parts of the code for convering json to bicep type, so I could re-use it. With change to System.Text.Json I'd need to rewrite them.
Plus, with using Jsonpath.net we are replacing one dependency to another - and jsonpath.net is a bit less popular than newtonsoft.json...

@anthony-c-martin , @majastrz - what's your opinion?

Reply:
That's a good point. Since the runtime has a hard dependency on Newtonsoft, it makes sense to use it for client-side functions as well.